### ISSUE TITLE
MinMetric produces incorrect result when used with MetricCollection and compute_groups=True

### ISSUE THREAD

**Issue by tshead2** [Lightning-AI/torchmetrics#1890]
## 🐛 Bug

Using `MinMetric` and `MaxMetric` in a `MetricCollection` with `compute_groups=True`, the minimum metric produces an incorrect result (it returns the maximum).

### To Reproduce

Steps to reproduce the behavior...

<details>

```
>>> import torchmetrics
>>> m = torchmetrics.MetricCollection({
...   "training/min": torchmetrics.MinMetric(),
...   "training/max": torchmetrics.MaxMetric(),
... }, compute_groups=True)
>>> m.update(1)
>>> m.update(2)
>>> m.compute()
{'training/max': tensor(2.), 'training/min': tensor(2.)}
```

</details>

### Expected behavior

The output from this example should be

```
{'training/max': tensor(2.), 'training/min': tensor(1.)}
```

... setting `compute_groups=False` produces the correct result.

### Environment

- TorchMetrics version 1.0.0, installed with pip
- Python 3.10.8, PyTorch 2.0.0, installed with conda
- Linux OS
